cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/OperationsDashboardHttpHandler.java

```diff
@@ -212,6 +215,11 @@ static DashboardProgramRunRecord runRecordToDashboardRecord(RunRecordMeta meta)
         // return "manual" if there's no trigger in the schedule info, but this should never happen
         .orElse(MANUAL);
     }
-    return new DashboardProgramRunRecord(runId, meta, meta.getArtifactId(), meta.getPrincipal(), startMethod);
+    String user = meta.getPrincipal();
+    if (user != null) {
```

Comment:
Can this be null? If yes then what will be user value which will be shown?

Reply:
yea, its null on non-kerberos cluster and its returned as null to user currently.

Reply:
should we say "cdap" or something else if its null ?

Reply:
@bdmogal, @gsps1 - what do you guys think?

Also, what to do in the case of an insecure environment can be taken up in a separate PR. This PR is just to fix so that the full principal isn't returned.